packages/base-manager/src/manager-base.ts

```diff
@@ -342,6 +361,197 @@ export abstract class ManagerBase implements IWidgetManager {
     return await modelPromise;
   }
 
+  /**
+   * Fetch all widgets states from the kernel using the control comm channel
+   * If this fails (control comm handler not implemented kernel side),
+   * it will fallback to `_loadFromKernelSlow`.
+   */
+  protected async _loadFromKernel(): Promise<void> {
```

Comment:
Can we add a note in the function documentation here that this is a utility method for subclasses to use as they see fit? Since we don't use this method in this base class, it is a bit confusing why it is defined here without extra context.

Reply:
(and same comment for the other loading function defined just below)

Reply:
Sure! Make sense. Thanks for the review!